pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -898,10 +901,12 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 	// Now validate all the listeners. Collate the tcp listeners first and then the HTTP listeners
 	// TODO: This is going to be bad for caching as the order of listeners in tcpListeners or httpListeners is not
 	// guaranteed.
+	invalid := 0
 	for name, l := range listenerMap {
 		if err := l.listener.Validate(); err != nil {
 			log.Warnf("buildSidecarOutboundListeners: error validating listener %s (type %v): %v", name, l.servicePort.Protocol, err)
-			invalidOutboundListeners.Add(1)
+			invalid++
+			stats.Record(context.Background(), invalidOutboundListeners.M(int64(invalid)))
```

Comment:
Is there a context available to pass down?
we seem to create context.Background here and elsewhere.

Reply:
Outdated. Resolving.

But, no, there did not seem to be any context passed around in this section of code.